pyccel/ast/core.py

```diff
@@ -2566,6 +2570,10 @@ def __init__(
             allocatable = False
         self.allocatable = allocatable
 
+        if is_const is None:
+            is_const = False
```

Comment:
It might be good to also add a type check

Reply:
i didn't get what you mean by adding a type check

Reply:
Checking that the type of the `is_const` argument is indeed a `bool`.

Reply:
But these two lines are a bit strange: the default value of this argument is `False`, so why should anybody pass `is_const = None` to this function?

Reply:
I see that the same (strange) approach is used for other arguments in this constructor, so I would suggest you try and clean it up a bit.